  0%|                                                                                                                                        | 0/2032 [00:00<?, ?it/s]/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/_functions.py:68: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
  0%|â–Ž                                                                                                                               | 4/2032 [00:07<51:28,  1.52s/it]Traceback (most recent call last):
  File "/home/hdiddee/coursework/sentiment-cartography/train.py", line 49, in <module>
    ).train()
      ^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/transformers/trainer.py", line 1938, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/transformers/trainer.py", line 2279, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/transformers/trainer.py", line 3318, in training_step
    loss = self.compute_loss(model, inputs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/transformers/trainer.py", line 3363, in compute_loss
    outputs = model(**inputs)
              ^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/data_parallel.py", line 187, in forward
    return self.gather(outputs, self.output_device)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/data_parallel.py", line 204, in gather
    return gather(outputs, output_device, dim=self.dim)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/scatter_gather.py", line 109, in gather
    res = gather_map(outputs)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/scatter_gather.py", line 100, in gather_map
    return type(out)((k, gather_map([d[k] for d in outputs]))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 9, in __init__
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/transformers/utils/generic.py", line 390, in __post_init__
    for idx, element in enumerate(iterator):
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/scatter_gather.py", line 100, in <genexpr>
    return type(out)((k, gather_map([d[k] for d in outputs]))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/scatter_gather.py", line 94, in gather_map
    return Gather.apply(target_device, dim, *outputs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/autograd/function.py", line 574, in apply
    return super().apply(*args, **kwargs)  # type: ignore[misc]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/_functions.py", line 75, in forward
    return comm.gather(inputs, ctx.dim, ctx.target_device)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hdiddee/environments/test/lib/python3.12/site-packages/torch/nn/parallel/comm.py", line 235, in gather
    return torch._C._gather(tensors, dim, destination)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 12.27 GiB. GPU 0 has a total capacity of 47.43 GiB of which 4.04 GiB is free. Including non-PyTorch memory, this process has 43.38 GiB memory in use. Of the allocated memory 29.59 GiB is allocated by PyTorch, and 13.37 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
